compiler/pipes/gen-tree-postprocess.cpp

```diff
@@ -89,6 +89,46 @@ VertexAdaptor<op_require> make_require_once_call(SrcFilePtr lib_main_file, Verte
   return req_once;
 }
 
+VertexPtr replace_call_func(VertexAdaptor<op_func_call> call, std::string func_name) {
+  auto new_call = VertexAdaptor<op_func_call>::create().set_location(call);
+  new_call->set_string(func_name);
+  return new_call;
+}
+
+VertexPtr process_microtime(VertexAdaptor<op_func_call> call) {
+  const auto &args = call->args();
+
+  // microtime()      -> microtime_string()
+  // microtime(false) -> microtime_string()
+  if (args.empty() || args[0]->type() == op_false) {
+    return replace_call_func(call, "microtime_string");
+  }
+
+  // microtime(true) -> microtime_float()
+  if (GenTree::get_actual_value(args[0])->type() == op_true) {
+    return replace_call_func(call, "microtime_float");
+  }
+
+  return call;
+}
+
+VertexPtr process_hrtime(VertexAdaptor<op_func_call> call) {
+  const auto &args = call->args();
+
+  // hrtime()      -> hrtime_array()
+  // hrtime(false) -> hrtime_array()
+  if (args.empty() || args[0]->type() == op_false) {
+    return replace_call_func(call, "hrtime_array");
+  }
+
+  // hrtime(true) -> hrtime_int()
+  if (GenTree::get_actual_value(args[0])->type() == op_true) {
+    return replace_call_func(call, "hrtime_int");
+  }
+
+  return call;
+}
+
```

Comment:
It's not okay to replace func calls in GenTreePostprocessPass — for many reasons. For example, 
1) if someone uses CONST_TRUE instead of true, it can't be done here
2) if someone passes wrong arguments count, we want an error to point to original function
3) we need to check all function names here, not only extern calls
4) some more

I suggest to look here: https://github.com/VKCOM/kphp/pull/510/files
Search for `maybe_replace_extern_func_call`: it's exactly what you need.
You can copy that function (dropping json from there), a comment above, and how it's called to your branch and write a body. That PR would be rebased onto master after merging yours.
Btw, in my branch with generics I also use that function.